### ISSUE TITLE
async / await function inside promise incorrectly requires resolve

### ISSUE THREAD

**Issue by RichardJECooke** [Microsoft/TypeScript#9179]
For bug reports, please include the information below.
__________________________________________________________ -->

**TypeScript Version:**  
1.8

```Typescript
const result = new Promise<boolean>( async (resolve, reject) => 
{
	const isValid = await this._bouncer.isTokenValid(token, resolve); //this function calls resolve(false)
	if (!isValid)
        return;
    resolve(true);
});
```

The above has no errors in atom-typescript, but when compiling I get this incorrect error:
`error TS7030: Not all code paths return a value`. I.e. Typescript doesn't know that `resolve` has been called elsewhere so it demands I call it right here in this code.

I can workaround the problem with this hack of swapping `return` with `throw null`

**Comment by mhegazy** [comment]
Atom is running the latest TS drop and not TS 1.8. you can get the same behavior using `typescript@next`. this issue has been fixed in latest, see https://github.com/Microsoft/TypeScript/issues/7446
